src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from "react";
+import DataTable from "react-data-table-component";
+import PageHeader from "../../../components/Common/PageHeader";
+import { ExportToExcel } from "../../../components/Utilities/Table/ExportToExcel";
+import ReviewCommentMasterModal from "./Validation/ReviewCommentMasterModal";
+import { useDispatch, useSelector } from "react-redux";
+import { getReviewCommentMasterListThunk } from "../../../redux/services/testCases/reviewCommentMaster";
+import { customSearchHandler } from "../../../utils/customFunction";
+import TableLoadingSkelton from "../../../components/custom/loader/TableLoadingSkelton";
+import { Col, Row } from "react-bootstrap";
+
+function ReviewCommentMasterComponent() {
+  const dispatch = useDispatch();
+
+  // // redux state
+  const { reviewCommentMasetrList, isLoading } = useSelector(
+    (state) => state?.reviewCommentMaster
+  );
+
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredRevieCommentMasterList, setFilteredReviewCommentkMasterList] =
+    useState([]);
+
+  const [addEditReviewCommentModal, setAddEditReviewCommentModal] = useState({
+    type: "",
+    data: "",
+    open: false,
+  });
+
+  // Function to handle search button click
+  const handleSearch = () => {
+    const filteredList = customSearchHandler(
+      reviewCommentMasetrList,
+      searchValue
+    );
+    setFilteredReviewCommentkMasterList(filteredList);
+  };
+
+  // Function to handle reset button click
+  const handleReset = () => {
+    setSearchValue("");
+    setFilteredReviewCommentkMasterList(reviewCommentMasetrList);
+  };
+
+  // Update the useEffect to update the filtered list when reviewCommentMasetrList changes
+  useEffect(() => {
+    setFilteredReviewCommentkMasterList(reviewCommentMasetrList);
+  }, [reviewCommentMasetrList]);
+
+  // Function to handle search onchange
+  useEffect(() => {
+    handleSearch();
+  }, [searchValue]);
+
+  const columns = [
+    {
+      name: "Sr. No.",
+      selector: (row, index) => index + 1,
+      sortable: false,
+      width: "70px",
+    },
+    {
+      name: "Action",
+      selector: (row) => (
+        <i
+          className="icofont-edit text-primary cp"
+          onClick={() =>
+            setAddEditReviewCommentModal({
+              type: "EDIT",
+              data: row,
+              open: true,
+            })
+          }
+        />
+      ),
+      sortable: false,
+      width: "70px",
+    },
+
+    {
+      name: "Status",
+      selector: (row) => (row.is_active === 1 ? "Active" : "Deactive"),
```

Comment:
If active , text color should be green or else red

Reply:
badge added